### ISSUE TITLE
Investigate upgrading Eigen dependency

### ISSUE THREAD

**Issue by SeanCurtis-TRI** [flexible-collision-library/fcl#285]
Currently, fcl uses [Eigen 3.0.5](https://github.com/flexible-collision-library/fcl/blob/master/CMakeLists.txt#L113). We need to find out if upgrading to [current version (3.3)](https://bitbucket.org/eigen/eigen/branch/3.3) is viable and desirable. Certainly, it would give us access to `NumTraits<S>::Literal` (as discussed in #282).

**Comment by jmirabel** [comment]
I didn't realize `Literal` was only available from a recent version of Eigen. I think it is a bad idea to depend on a recent version of Eigen as it prevents many people from using FCL. IMHO, the availability of `Literal` is not a good enough reason.

**Comment by jmirabel** [comment]
Of course, it is a good idea to be *compatible* (but not *dependent*).
